docs/howto/features/binderhub-ui.md

```diff
@@ -0,0 +1,181 @@
+    loadRoles:
+      # The role binder allows service binder to start and stop servers
+      # and read (but not modify) any user’s model
+      binder:
+        services:
+          - binder
+        scopes:
+          - servers
+          - read:users # admin:users is required if authentication isn't enabled
+      # The role user allows access to the user’s own resources and to access
+      # only the binder service
+      user:
+        scopes:
+          - self
+          # Admin users will by default have access:services, so this is only
+          # observed to be required for non-admin users.
+          - access:services!service=binder
+    config:
+      BinderSpawnerMixin:
+        auth_enabled: true
+```
+
+## Setup the image registry
+
+Follow the guide at [](howto:features:imagebuilding-hub:image-registry) of the imagebuilding hub.
+
+## Setup the `binderhub-service` chart
+
+We will use the [binderhub-service](https://github.com/2i2c-org/binderhub-service/) Helm chart to run BinderHub, the Python software, as a standalone service to build and push images with [repo2docker](https://github.com/jupyterhub/repo2docker), next to JupyterHub.
+
+1. Setup the `binderhub-service` config
+
+    ```yaml
+    binderhub-service:
+      enabled: true
+      # enable the binderhub-service ingress and configure it
+      # so that we get a public domain for binderhub to run
+      ingress:
+        enabled: true
+        hosts:
+          - <binderhub-public-url>.2i2c.cloud
+        tls:
+          - secretName: binder-https-auto-tls
+            hosts:
+              - <binderhub-public-url>.2i2c.cloud
+      config:
+        BinderHub:
+          base_url: /
+          hub_url: https://<jupyterhub-public-url>.2i2c.cloud
+          badge_base_url: https://<binderhub-public-url>.2i2c.cloud
+          auth_enabled: true
+          enable_api_only_mode: false
+          # something like <region>-docker.pkg.dev/<project-name>/<repository-name> for grc.io
+          # or quay.io/org/repo/cluster-hub/ for quay.io
+          image_prefix: <repository_path>
+      buildPodsRegistryCredentials:
+        # registry server address like https://quay.io or https://us-central1-docker.pkg.dev
+        server: <server_address>
+        # robot account namer or "_json_key" if using grc.io
+        username: <account_name>
+      extraConfig:
+        # FIXME: set KubernetesBuildExecutor.push_secret again
+        #        without this for some reason the build pods
+        #        search after the wrong secret name (i.e. the default name)
+        #        set by binderhub in KubernetesBuildExecutor.push_secret
+        01-binderhub-service-set-push-secret: |
+          import os
+          c.KubernetesBuildExecutor.push_secret = os.environ["PUSH_SECRET_NAME"]
+      extraEnv:
+        - name: JUPYTERHUB_API_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: '{{ include "jupyterhub.hub.fullname" . }}'
```

Comment:
Let’s just say “hub” here insteaf of templating. It will never be a different value for us.

Reply:
This template name i also consider an internal implementation detail i don’t want to dive into and depend on